src/passes/OptimizeInstructions.cpp

```diff
@@ -2248,8 +2248,27 @@ struct OptimizeInstructions
     }
 
     if (curr->op == ExternConvertAny || curr->op == AnyConvertExtern) {
-      // We can't optimize these. Even removing a non-null cast is not valid as
-      // they allow nulls to filter through, unlike other RefAs*.
+      // These pass nulls through, and we can reorder them with null traps:
+      //
+      //  (ref.externalize/internalize (ref.as_non_null.. ))
+      // =>
+      //  (ref.as_non_null (ref.externalize/internalize ..))
```

Comment:
The instructions are now called `any.convert_extern` and `extern.convert_any`. It would be good to use those names in the comments for clarity and searchability.

Reply:
Good point, thanks, done.